src/components/Button/Button.js

```diff
@@ -93,14 +95,37 @@ function PrivateButton({
     ''
   )
 
+  const [isDebounced, setIsDebounced] = useState(false)
+
+  useEffect(() => {
+    if (isDebounced) {
+      setTimeout(() => setIsDebounced(false), debounceDurationMs)
+    }
+  }, [isDebounced])
+
+  let debouncedOnClick
```

Comment:
What is the behavior if we pass in `undefined` below? Do we want a default no-op?

Reply:
if value is undefined / some unexpected value, just passing through so it behaves as it would before this change https://github.com/getethos/ethos-design-system/blob/9516b4792afc44792efa5e20a22664767b4607bb/src/hooks/useDebounce.js#L23